pallas-miniprotocols/src/handshake/client.rs

```diff
@@ -1,27 +1,8 @@
 use pallas_codec::Fragment;
-use pallas_multiplexer::agents::{Channel, ChannelBuffer, ChannelError};
+use pallas_multiplexer::agents::{Channel, ChannelBuffer};
 use std::marker::PhantomData;
-use thiserror::*;
 
-use super::{Message, RefuseReason, State, VersionNumber, VersionTable};
-
-#[derive(Error, Debug)]
-pub enum Error {
-    #[error("attempted to receive message while agency is ours")]
-    AgencyIsOurs,
-
-    #[error("attempted to send message while agency is theirs")]
-    AgencyIsTheirs,
-
-    #[error("inbound message is not valid for current state")]
-    InvalidInbound,
-
-    #[error("outbound message is not valid for current state")]
-    InvalidOutbound,
-
-    #[error("error while sending or receiving data through the channel")]
-    ChannelError(ChannelError),
-}
+use super::{Message, RefuseReason, State, VersionNumber, VersionTable, Error};
 
 #[derive(Debug)]
 pub enum Confirmation<D> {
```

Comment:
One thing I noticed is this `Confirmation` type uses the term `Reject` while everything else uses `Refuse`; do we want to make it consistent?